SonarQube.Client/SonarQubeService.cs

```diff
@@ -103,12 +103,22 @@ public SonarQubeService(HttpMessageHandler messageHandler, string userAgent, ILo
         /// <param name="configure">Action that configures a type instance that implements TRequest.</param>
         /// <param name="token">Cancellation token.</param>
         /// <returns>Returns the result of the request invocation.</returns>
-        protected virtual async Task<TResponse> InvokeRequestAsync<TRequest, TResponse>(Action<TRequest> configure,
+        private Task<TResponse> InvokeCheckedRequestAsync<TRequest, TResponse>(Action<TRequest> configure,
             CancellationToken token)
             where TRequest : IRequest<TResponse>
         {
             EnsureIsConnected();
 
+            return InvokeUncheckedRequestAsync<TRequest, TResponse>(configure, token);
+        }
+
+        /// <summary>
+        /// Executes the call without checking whether the connection to the server has been established. This should only normally be used directly while connecting.
+        /// Other uses should call <see cref="InvokeCheckedRequestAsync{TRequest,TResponse}(System.Threading.CancellationToken)"/>.
+        /// </summary>
+        protected virtual async Task<TResponse> InvokeUncheckedRequestAsync<TRequest, TResponse>(Action<TRequest> configure, CancellationToken token)
```

Comment:
@duncanp-sonar you meant changing this method to virtual, right?

Reply:
Yep; it's now the inner-most call that actually does the work.